src/servers/src/prom.rs

```diff
@@ -699,3 +704,115 @@ fn promql_expr_to_metric_name(expr: &PromqlExpr) -> Option<String> {
         }
     }
 }
+
+#[derive(Debug, Default, Serialize, Deserialize, JsonSchema)]
+pub struct LabelValueQuery {
+    start: Option<String>,
+    end: Option<String>,
+    #[serde(flatten)]
+    matches: Matches,
+    db: Option<String>,
+}
+
+#[axum_macros::debug_handler]
+pub async fn label_values_query(
+    State(handler): State<PromHandlerRef>,
+    Path(label_name): Path<String>,
+    Query(params): Query<LabelValueQuery>,
+) -> Json<PromJsonResponse> {
+    let queries = params.matches.0;
+    if queries.is_empty() {
+        return PromJsonResponse::error("Invalid argument", "match[] parameter is required");
+    }
+
+    let start = params.start.unwrap_or_else(yesterday_rfc3339);
+    let end = params.end.unwrap_or_else(current_time_rfc3339);
+    let db = &params.db.unwrap_or(DEFAULT_SCHEMA_NAME.to_string());
+    let (catalog, schema) = crate::parse_catalog_and_schema_from_client_database_name(db);
+    let query_ctx = Arc::new(QueryContext::with(catalog, schema));
+
+    let mut label_values = HashSet::new();
+
+    for query in queries {
+        let prom_query = PromQuery {
+            query,
+            start: start.clone(),
+            end: end.clone(),
+            // TODO(ccl): find a better value for step
+            step: DEFAULT_LOOKBACK_STRING.to_string(),
```

Comment:
Looks like the `labels` API doesn't support `step`, so the `TODO` is unnecessary.

Reply:
But `PromQuery` requires `step`, different `step` may produce different query result?

Reply:
I mean we can always use the `DEFAULT_LOOKBACK_STRING`. The `TODO` comment is unnecessary.